### PR TITLE
Dont encode to utf8 for XML and JSON since they expect unicode

### PR DESCRIPTION
**Pull request by nicpottier**

### CODE REVIEW

File: temba/channels/models.py [hunk 1]
```diff
@@ -1181,11 +1181,11 @@ def sync_channel(cls, gcm_id, channel=None):  # pragma: no cover
...
-            replacement = six.text_type(variables[key]).encode('utf-8')
+            replacement = six.text_type(variables[key])
 
             # encode based on our content type
             if content_type == Channel.CONTENT_TYPE_URLENCODED:
-                replacement = quote_plus(replacement)
+                replacement = quote_plus(replacement.encode('utf8'))
 
             # if this is JSON, need to wrap in quotes (and escape them)
             elif content_type == Channel.CONTENT_TYPE_JSON:
```
Comment (norkans7) [inline temba/channels/models.py]: What about the JSON content type? I think it should be encode to utf8

Comment (nicpottier) [inline temba/channels/models.py]: That's taken care of by the requests library. The text is unicode at this point, encoding changes it to bytes, which is what quote_plus expects, but not what the other pieces need.

Comment (rowanseymour) [inline temba/channels/models.py]: Django comes with its own version of quote_plus (`django.utils.http.urlquote_plus`) which can take unicode

Comment (nicpottier) [inline temba/channels/models.py]: Oh nice, will switch to that.

**Comment by nicpottier** [comment]
can i get a thumbs up!?

**Pull request merged**